util/git/client.go

```diff
@@ -405,6 +406,43 @@ func (m *nativeGitClient) LsRemote(revision string) (res string, err error) {
 	return
 }
 
+// LsRemote resolves the commit SHA of a specific branch, tag, or HEAD. If the supplied revision
+// does not resolve, and "looks" like a 7+ hexadecimal commit SHA, it return the revision string.
+// Otherwise, it returns an error indicating that the revision could not be resolved. This method
+// runs with in-memory storage and is safe to run concurrently, or to be run without a git
+// repository locally cloned.
+func (m *nativeGitClient) LsRemoteGitCLI(revision string) (res string, err error) {
+	for attempt := 0; attempt < maxAttemptsCount; attempt++ {
+		if res, err = m.lsremoteGitCLI(revision); err == nil {
+			return
+		}
+	}
+	return
+}
+
+// ls-remote using git cli
+func (m *nativeGitClient) lsremoteGitCLI(revision string) (string, error) {
+	if revision == "" {
+		revision = "HEAD"
+	}
+	out, err := m.runCredentialedCmdWithOutput("git", "ls-remote", "origin", revision)
+	if err != nil {
+		log.Error(err)
+		return "", err
+	}
+	if out != "" {
+		rev := strings.Fields(out)
+		if !strings.Contains(revision, rev[len(rev)-1]) {
```

Comment:
Rather than using Contains on the full string, I would suggest only running equals against the data after the last slash, eg:
```
refs/heads/|add-rate-limiter|   (compare the value between the |s)
```